renter/session.go

```diff
@@ -0,0 +1,245 @@
+		AccountID: accountID,
+	})
+	if err != nil {
+		return types.ZeroCurrency, fmt.Errorf("failed to write account balance request: %w", err)
+	}
+
+	var resp rhp.RPCFundAccountResponse
+	if err := rpc.ReadResponse(stream, &resp); err != nil {
+		return types.ZeroCurrency, fmt.Errorf("failed to read account balance response: %w", err)
+	}
+
+	return resp.Balance, nil
+}
+
+// LatestRevision returns the latest revision of a contract.
+func (s *Session) LatestRevision(contractID types.ElementID, payment PaymentMethod) (types.FileContractRevision, error) {
+	if payment == nil {
+		return types.FileContractRevision{}, errors.New("payment method is required")
+	}
+
+	stream, err := s.session.DialStream()
+	if err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to open new stream: %w", err)
+	}
+	defer stream.Close()
+
+	id, settings, err := s.currentSettings()
+	if err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to load price table: %w", err)
+	}
+
+	if err := rpc.WriteRequest(stream, rhp.RPCLatestRevisionID, &id); err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to write latest revision request: %w", err)
+	}
+
+	if err := payment.Pay(stream, settings.RPCLatestRevisionCost); err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to pay for latest revision: %w", err)
+	}
+
+	req := &rhp.RPCLatestRevisionRequest{
+		ContractID: contractID,
+	}
+	if err := rpc.WriteResponse(stream, req); err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to write latest revision request: %w", err)
+	}
+
+	var resp rhp.RPCLatestRevisionResponse
+	if err := rpc.ReadResponse(stream, &resp); err != nil {
+		return types.FileContractRevision{}, fmt.Errorf("failed to read latest revision response: %w", err)
+	}
+	return resp.Revision, nil
+}
+
+// Settings returns the current settings for the host. If a payment method is
+// provided the setting UID should be used when interacting with the host.
+func (s *Session) Settings(payment PaymentMethod) (settings rhp.HostSettings, _ error) {
+	stream, err := s.session.DialStream()
+	if err != nil {
+		return rhp.HostSettings{}, fmt.Errorf("failed to open stream: %w", err)
+	}
+	defer stream.Close()
+
+	if err := rpc.WriteRequest(stream, rhp.RPCSettingsID, nil); err != nil {
+		return rhp.HostSettings{}, fmt.Errorf("failed to write settings request: %w", err)
+	}
+
+	var resp rhp.RPCSettingsResponse
+	if err := rpc.ReadResponse(stream, &resp); err != nil {
+		return rhp.HostSettings{}, fmt.Errorf("failed to read settings response: %w", err)
+	}
+
+	if err := json.Unmarshal(resp.Settings, &settings); err != nil {
+		return rhp.HostSettings{}, fmt.Errorf("failed to decode settings: %w", err)
+	}
+
+	if payment == nil {
+		return
+	} else if err := payment.Pay(stream, settings.RPCHostSettingsCost); err != nil {
+		return rhp.HostSettings{}, fmt.Errorf("failed to pay for settings: %w", err)
+	}
```

Comment:
maybe we should just have two separate `Session` methods, one for paid settings, one for unpaid

Reply:
(or two separate RPCs, even!)

Reply:
Maybe `ScanSettings` and `RegisterSettings`?  The API and implementation are pretty clean, though; I can't think of a good reason to break it up.

Reply:
We had talked about getting rid of settings registration entirely, which would make the Settings RPC unpaid anyway.

Reply:
yeah, probably don't need a separate RPC, but let's have separate session methods.